signer/src/bitcoin/utxo.rs

```diff
@@ -160,16 +162,24 @@ impl SbtcRequests {
             })
             .map(RequestRef::Withdrawal);
 
-        // Now we filter deposit requests where the user's max fee could
-        // be less than the fee we may charge. This is simpler because
-        // deposit UTXOs have a known fixed size.
+        // Filter deposit requests based on two constraints:
+        // 1. The user's max fee must be >= our minimum required fee for deposits
+        //     (based on fixed deposit tx size)
+        // 2. The total amount being minted must stay under the maximum allowed mintable amount
         let minimum_deposit_fee = self.compute_minimum_fee(SOLO_DEPOSIT_TX_VSIZE);
-        let deposits = self
-            .deposits
-            .iter()
-            .filter(|req| req.max_fee >= minimum_deposit_fee)
-            .map(RequestRef::Deposit);
 
+        let mut amount_to_mint = 0;
+        let deposits = self.deposits.iter().filter_map(|req| {
+            let valid_fee = req.max_fee >= minimum_deposit_fee;
+            let valid_amount = amount_to_mint + req.amount <= self.max_mintable;
+
+            if valid_fee && valid_amount {
+                amount_to_mint += req.amount;
```

Comment:
The actual minting amount can be greater than this, right? See my reply to your PR comment. I _think_ we need to check `amount + (max_fee - min_deposit_fee)`